Parsers/ASimNetworkSession/Parsers/vimNetworkSessionCheckPointFirewall.yaml

```diff
@@ -0,0 +1,329 @@
+            dstipaddr_has_any_prefix:dynamic=dynamic([]), 
+            ipaddr_has_any_prefix:dynamic=dynamic([]),
+            dstportnumber:int=int(null), 
+            hostname_has_any:dynamic=dynamic([]), 
+            dvcaction:dynamic=dynamic([]), 
+            eventresult:string='*', 
+            disabled:bool=false)
+        {
+        let src_or_any = set_union(srcipaddr_has_any_prefix, ipaddr_has_any_prefix); 
+        let dst_or_any = set_union(dstipaddr_has_any_prefix, ipaddr_has_any_prefix);
+        CommonSecurityLog
+        | where not(disabled)
+        | where
+                        (isnull(starttime) or TimeGenerated >= starttime) 
+                        and (isnull(endtime) or TimeGenerated <= endtime)
+        | where 
+                        array_length(hostname_has_any) == 0
+        | where DeviceVendor=="Check Point" and DeviceProduct=="VPN-1 & FireWall-1"
+        | where (isnull(dstportnumber) or (DestinationPort == dstportnumber))
+        | where ((eventresult == "*") 
+                        or (eventresult == "Success" and AdditionalExtensions has "Accept") 
+                        or (eventresult == "Failure" and AdditionalExtensions !has "Accept"))
+        | extend temp_isSrcMatch=has_any_ipv4_prefix(SourceIP,src_or_any), 
+                 temp_isDstMatch=has_any_ipv4_prefix(DestinationIP,dst_or_any)
+        | extend ASimMatchingIpAddr = case(
+                            array_length(src_or_any) == 0 and array_length(dst_or_any) == 0, "-", // match not requested
+                            (temp_isSrcMatch and temp_isDstMatch), "Both", // has to be checked before the individual 
+                            temp_isSrcMatch, "SrcIpAddr",
+                            temp_isDstMatch, "DstIpAddr",
+                            "No match"
+                        )
+        | where ASimMatchingIpAddr != "No match"
+        | extend 
+                EventProduct = "Firewall",
+                EventCount = toint(1),
+                EventType = "NetworkSession",
+                EventSchema = "NetworkSession",
+                EventSchemaVersion = "0.2.4"
+        | parse-kv AdditionalExtensions as (
+                    rule_action:string, 
+                    rule_uid:string,
+                    loguid:string,
+                    origin:string,
+                    originsicname:string,
+                    inzone:string,
+                    outzone:string,
+                    conn_direction:string,
+                    alert:string,
+                    inspection_category:string,
+                    inspection_item:string
+                ) with (pair_delimiter=';', kv_delimiter='=')
+        | extend
+                ThreatCategory = coalesce(alert, inspection_category),
+                NetworkRuleName = coalesce(DeviceCustomString2, rule_uid, Activity),
+                EventStartTime = TimeGenerated,
+                DvcHostname = tostring(extract_all(@"CN\\=(\S+),",originsicname)[0])
```

Comment:
Why extract_all and not extract if you have just one item? also, I think that parse will work here.

Reply:
Replaced with parse